### PR TITLE
feat: merge v2.14.0 into develop

### PR DESCRIPTION
**Pull request by RetricSu**
## New Content
- Add a new Section "History & Hard Fork" including 4 docs:
   - Intro to Hard Fork
   - CKB Hard Fork History
   - History VM Version
   - Rethinking Forks
- Add new content for Script sections:
   - Add a new doc about Spawn direct cross-script calling
   - Add a new Spawn tutorial
   - Add a new doc about vm selection mechanism
   - Add a new doc about Script upgrade workflow
   - Update Script debug docs with multiple methods
   - Update ckb js vm tutorial with Meepo Hard Fork version and offckb repl
- Add new docs about CKB transaction fee estimator feature
- Add installation guide & update tutorial overview
- Update the definition of hash_type for Meepo Hard Fork

## Fixes
- fix: redirect deadlink of /docs/basics/tools#neuron-wallet

## Others
- Clarify the minimal version of CCC used across the doc site
- Split cli-tool into single page doc and bump offckb v0.3.0
- Adjust the child docs order in the Script section

### CODE REVIEW

File: website/docs/history-and-hard-forks/rethinking-fork.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+id: rethinking-forks
+title: "Rethinking Forks"
+---
+
+# Rethinking Forks: The Philosophy Behind CKB's Network Upgrade Design
+
+> Originally wrote by Jan Xie
```
Comment (linnnsss) [inline website/docs/history-and-hard-forks/rethinking-fork.mdx:8]:
```suggestion
> Originally written by Jan Xie
```

File: website/sidebars.js [hunk 2]
```diff
@@ -122,12 +129,25 @@ export default {
...
+        "history-and-hard-forks/rethinking-forks",
+      ],
+    },
     {
       type: "category",
       label: "Tech Explanation",
       className: "category-tech-explanation",
       collapsible: false,
       items: [
+        "tech-explanation/fee-estimator",
```
Comment (yfeng2824) [inline website/sidebars.js:150]: Should we move this below "Transaction"?

File: website/docs/history-and-hard-forks/history-vm-version.mdx [hunk 3]
```diff
@@ -0,0 +1,212 @@
+---
+id: history-vm-version
+title: "History VM Version"
```
Comment (yfeng2824) [inline website/docs/history-and-hard-forks/history-vm-version.mdx:3]: Can we rename it to "VM Version History"?

File: website/docs/tech-explanation/ckb-vm.md [hunk 4]
```diff
@@ -17,6 +17,10 @@ Any programming language that can target RISC-V can be used natively for develop
 
 Nervos CKB offers native SDKs in several mainstream programming languages, such as JavaScript, Rust, Go, Java, and Ruby.
 
+## CKB-VM Version
+
+The CKB network has introduced various CKB-VM versions over time to enhance security, performance, resolve bugs, and support new RISC-V extensions. To check out all versions of the CKB-VM from previous hard fork events, please visit [History VM Version](/docs/history-and-hard-forks/history-vm-version)
```
Comment (yfeng2824) [inline website/docs/tech-explanation/ckb-vm.md:22]: "History VM Version" -> “VM Version History”

File: website/docs/script/spawn-direct-cross-script-calling.mdx [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+      +----------+
+----->+ Script A +--------------+----------->
+      +-----+----+              ^
+            |                   |
+            | Spawn             | Exit
+            v                   |
+      +----------+              |
+      | Script B +--------------+
+      +----------+
+```
```
Comment (yfeng2824) [inline website/docs/script/spawn-direct-cross-script-calling.mdx:36]: It seems that these ASCII text diagrams appear misaligned in the preview, despite using a monospace font (space mono).  
Screenshot 2024-12-17 at 10 36 02 AM

File: website/sidebars.js (hunk 2 above)
Comment (RetricSu) [reply website/sidebars.js:150]: yeah in the long term, but right now this is a new content, so i would like to make it first for a while

**Review by linnnsss — APPROVED** [review]

File: website/docs/script/spawn-direct-cross-script-calling.mdx (hunk 5 above)
Comment (RetricSu) [reply website/docs/script/spawn-direct-cross-script-calling.mdx:36]: image
it seems fine on my brave browser. what is your environment?

**Pull request merged**